### PR TITLE
Traffic Endpoints Support

### LINKED ISSUE #926: GitHub Repo Traffic Endpoints are Missing
Hello there,

Love the work that's being done here.

It appears that there is a set of endpoints that are from the traffic endpoints.

Specifically
`GET /repos/:owner/:repo/traffic/popular/referrers`
`GET /repos/:owner/:repo/traffic/popular/paths`
`GET /repos/:owner/:repo/traffic/views`
`GET /repos/:owner/:repo/traffic/clones`

These can be found here:
https://developer.github.com/v3/repos/traffic/

Would love to see these added!

Keep up the great work!

### PR DESCRIPTION
**Pull request by jkufro**
Added support for the traffic endpoints found at https://developer.github.com/v3/repos/traffic/.

We added four functions to the Repository class (one for each of the endpoints) with corresponding unit tests. 

Fixes #926

### CODE REVIEW

File: doc/examples/Repository.rst [hunk 1]
```diff
@@ -138,6 +138,40 @@ Delete a file in the repository
     >>> repo.delete_file(contents.path, "remove test", contents.sha, branch="test")
     {'commit': Commit(sha="0f40b0b4f31f62454f1758d7e6b384795e48fd96"), 'content': NotSet}
 
+Get the top 10 referrers over the last 14 days
```
Comment (sfdye) [inline doc/examples/Repository.rst]: Maybe adding a few example output here?

**Comment by sfdye** [comment]
Left a few comments. Great team efforts by the way 👍

File: github/Repository.py [hunk 2]
```diff
@@ -1470,17 +1479,93 @@ def get_file_contents(self, path, ref=github.GithubObject.NotSet):
...
+                for item in data
+            ]
+
+    def get_views_traffic(self, per=github.GithubObject.NotSet):
+        """
+        :calls: `GET /repos/:owner/:repo/traffic/views <https://developer.github.com/v3/repos/traffic/>`_
+        :param per: string, must be one of day or week, day by default
+        :rtype: None or list of :class:`github.View.View`
+        """
+        assert per is github.GithubObject.NotSet or (isinstance(per, (str, unicode)) and (per is "day" or per is "week")), "per must be day or week, day by default"
```
[the remark this answers is not in the document]
Comment (jkufro) [reply github/Repository.py]: [PEP8 says that comparisons to singletons should use `is` over `==`](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)

[the remark this answers is not in the document]
Comment (sfdye) [reply github/Repository.py]: I am saying string comparisons. You are comparing the string values, not the object itself.

**Comment by jkufro** [comment]
Thanks for your feedback. I made the changes you requested in my most recent commit.

**Comment by sfdye** [comment]
Awesome, thank you all 👍

**Pull request merged**